src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/SecondarySite/index.js

```diff
@@ -29,154 +29,219 @@ const StepPrimarySetup = () => {
+		if ( siteClassificationsData?.body.length === 0 ) {
+			siteClassificationsData = fallbackDataTypes;
+		}
+		setSiteClassData( siteClassificationsData?.body );
+		setPrimaryTypeList(
+			Object.keys( siteClassificationsData?.body?.types )
+		);
 
-	/**This condition fills the data in input box if the saved category isn't a subcategory from the content*/
-	if (
-		selectedCategoryInStore &&
-		! inputCategVal &&
-		subCategories.indexOf( selectedCategoryInStore ) === -1
-	) {
-		if ( selectedCategoryInStore !== 'secondaryCategory' )
-			changeInputCateg( selectedCategoryInStore );
-	}
+		if ( currentData?.data?.siteType?.primary !== '' )
+			changePrimaryType( currentData?.data?.siteType?.primary );
+		changeSecondaryType( currentData?.data?.siteType?.secondary ?? '' );
+
+		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
+			changePrimaryType( defaultPrimaryType );
+			changeSecondaryType( '' );
+			changeInputCateg( currentData?.data?.siteType?.secondary );
+		}
+
+		if ( currentData?.data?.siteType?.labelSec === 'custom' ) {
+			changePrimaryType( defaultPrimaryType );
+			changeSecondaryType( '' );
+			changeInputCateg( currentData?.data?.siteType?.secondary );
+		}
+	};
 
 	/**
 	 * Function which saves data in redux when category name is put-in via input box
 	 *
-	 * @param  input
+	 * @param {string} input
 	 */
 	const categoryInput = ( input ) => {
-		changeCategory( -1 );
+		changeSecondaryType( '' );
 		changeInputCateg( input?.target?.value );
 		const currentDataCopy = currentData;
+		currentDataCopy.data.siteType.labelSec = 'custom';
 		currentDataCopy.data.siteType.secondary = input?.target?.value;
 		setCurrentOnboardingData( currentDataCopy );
 	};
 
 	/**
 	 * Function which saves data in redux when category name is chosen via categories displayed
 	 *
-	 * @param  idxOfElm
+	 * @param {string} secType
 	 */
-	const handleCategoryClick = ( idxOfElm ) => {
-		changeCategory( idxOfElm );
+	const handleCategoryClick = ( secType ) => {
+		changeSecondaryType( secType );
 		changeInputCateg( '' );
 		const currentDataCopy = currentData;
-		currentDataCopy.data.siteType.secondary =
-			categoriesArray[ 0 ]?.subCategories[ idxOfElm ];
+		currentDataCopy.data.siteType.labelPri = '';
+		currentDataCopy.data.siteType.labelSec = '';
+		currentDataCopy.data.siteType.primary = primaryType;
+		currentDataCopy.data.siteType.secondary = secType;
 		setCurrentOnboardingData( currentDataCopy );
 	};
 
+	const changePrimaryPrev = () => {
+		const idx = primaryTypeList.findIndex( ( val ) => primaryType === val );
+
+		if ( idx === 0 )
+			changePrimaryType( primaryTypeList[ primaryTypeList.length - 1 ] );
+		else changePrimaryType( primaryTypeList[ idx - 1 ] );
+	};
+
+	const changePrimaryNext = () => {
+		const idx = primaryTypeList.findIndex( ( val ) => primaryType === val );
+
+		if ( idx === primaryTypeList.length - 1 )
+			changePrimaryType( primaryTypeList[ 0 ] );
+		else changePrimaryType( primaryTypeList[ idx + 1 ] );
+	};
```

Comment:
Can we move this to a single function and pass `action` as a parameter? When doing that, we can simplify some of this if else logic as well?